apps/src/lib/script-editor/FlexGroup.jsx

```diff
@@ -0,0 +1,219 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import _ from 'lodash';
+import color from '../../util/color';
+import {borderRadius, ControlTypes} from './constants';
+import OrderControls from './OrderControls';
+import StageCard from './StageCard';
+import {NEW_LEVEL_ID, addStage, addGroup} from './editorRedux';
+import {LevelKind} from '@cdo/apps/util/sharedConstants';
+
+const styles = {
+  groupHeader: {
+    fontSize: 18,
+    color: 'white',
+    background: color.cyan,
+    borderTopLeftRadius: borderRadius,
+    borderTopRightRadius: borderRadius,
+    padding: 10
+  },
+  groupBody: {
+    background: color.lightest_cyan,
+    borderBottomLeftRadius: borderRadius,
+    borderBottomRightRadius: borderRadius,
+    padding: 10,
+    marginBottom: 20
+  },
+  addGroup: {
+    fontSize: 14,
+    color: 'white',
+    background: color.cyan,
+    border: `1px solid ${color.cyan}`,
+    boxShadow: 'none',
+    margin: '0 0 30px 0'
+  },
+  addStage: {
+    fontSize: 14,
+    color: '#5b6770',
+    background: 'white',
+    border: '1px solid #ccc',
+    boxShadow: 'none',
+    margin: '0 10px 10px 10px'
+  }
+};
+
+class FlexGroup extends Component {
+  static propTypes = {
+    addGroup: PropTypes.func.isRequired,
+    addStage: PropTypes.func.isRequired,
+    stages: PropTypes.array.isRequired,
+    levelKeyList: PropTypes.object.isRequired
+  };
+
+  handleAddGroup = () => {
+    this.props.addGroup(
+      prompt('Enter new stage name'),
+      prompt('Enter new group name')
+    );
+  };
+
+  handleAddStage = position => {
+    this.props.addStage(position, prompt('Enter new stage name'));
+  };
+
+  /**
+   * Generate the ScriptDSL format.
+   * @param stages
+   * @return {string}
+   */
+  serializeStages = stages => {
```

Comment:
I think we might want to move the serialization logic into redux, to make it easier to test. I don't see an obvious way to avoid ultimately rendering into a react component so that it gets picked up by the form submission, but I think that's ok since it doesn't seem to be a performance problem.